libmattermost.c

```diff
@@ -1016,12 +1078,17 @@ mm_add_channels_to_blist(MattermostAccount *ma, JsonNode *node, gpointer user_da
 	guint i, len = json_array_get_length(channels);
 	PurpleGroup *default_group = mm_get_or_create_default_group();
 	GList *ids = NULL;
-	
+	GList *seen_ids = NULL;
+
 	for (i = 0; i < len; i++) {
 		JsonObject *channel = json_array_get_object_element(channels, i);
 		const gchar *id = json_object_get_string_member(channel, "id");
 		const gchar *name = json_object_get_string_member(channel, "display_name");
 		const gchar *room_type = json_object_get_string_member(channel, "type");
+
+		if(id && *id) {
+			seen_ids=g_list_append(seen_ids,g_strdup(id));

```

Comment:
Using `g_list_prepend()` is generally faster so that the whole GList doesn't have to be stepped through to add to the end